### PR TITLE
Issue #7687: Adding code examples for ImportOrder check

### PR DESCRIPTION
**Pull request by aryaniiit002**
Issue #7687: Adding code examples for ImportOrder check
minor: fix incorrect property reference (#7687)

### CODE REVIEW

**Comment by aryaniiit002** [comment]
@nmancus1 please check if everything okay.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 1]
```diff
@@ -239,7 +309,7 @@
  * </p>
  * <pre>
  * &lt;module name=&quot;ImportOrder&quot;&gt;
- *   &lt;property name=&quot;groups&quot; value=&quot;*,javax,java&quot;/&gt;
+ *   &lt;property name=&quot;groups&quot; value=&quot;*,java,javax&quot;/&gt;
```
Comment (nmancus1) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: Why was this changed?

**Review by nmancus1 — COMMENTED** [review]
Just one question:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 1 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: When i was testing this config with code example i was getting wrong order at line 329 i thought it was bcoz of top static import but it was also bcoz of this value ordered javax then java. So i could either change the order in code example Or here, i did it here.

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: When i was testing this config with below code example i was getting wrong order violation at line 392 i thought it was only bcoz of above static import but it was also bcoz of this valued ordered javax then java so to correct either i change the imports or this value here. I changed here

Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: I think it would be better to keep `java, javax` and change the example; `java, javax` is the convention followed by many organizations and IDE's.  Remember, users often copy and paste directly from examples.

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: ugh, if you keep` java then javax` then there is no need to change code.

Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: >  i was getting wrong order violation at line 392

line 392 of what file? I am confused by your comment. The code example on line 392 of `ImportOrderCheck.java` is not associated with this config.

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: i am so sorry i wasted a lot of time it's not 392 it's 329 sorry

Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: Thanks for clarifying that.  Please come up with some way to keep `java, javax` in config but still show good violation examples.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 2]
```diff
@@ -251,6 +321,22 @@
  * &lt;/module&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * import static javax.swing.WindowConstants.*; // Violation, All static imports comes at bottom
+ *
```
Comment (aryaniiit002) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:328]: i think removing 328 empty line and violation comment at 329 will do the job bcoz that violation as you know due to static import line at 327 and if you remember i asked you to keep 327 to make it simple but if that's not working we can remove it

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 3]
```diff
@@ -251,6 +321,22 @@
...
+ * import java.net.URL; // Violation, Wrong order for 'java.net.URL'
```
Comment (aryaniiit002) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:329]: i think removing 328 empty line and violation comment at 329 will do the job bcoz that violation as you know due to static import line at 327 and if you remember i asked you to keep 327 to make it simple but if that's not working we can remove it

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 1 above)
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: Thanks for clarifying, I was confused by "code example"; you were referring to the configuration. I see now.

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: It's may be bcoz of my english sorry for that. 
So now should i make any changes here?

Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: I think this would be a good example that is representative of the intellij default import settings:

```bash

➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
        <module name="ImportOrder">
            <property name="groups" value="*,javax,java"/>
            <property name="ordered" value="true"/>
            <property name="separated" value="false"/>
            <property name="option" value="bottom"/>
            <property name="sortStaticImportsAlphabetically" value="true"/>
        </module>
        <module name="SuppressionXpathSingleFilter">
            <property name="checks" value="ImportOrder"/>
            <property name="message" value="^'java\..*'.*"/>
        </module>
    </module>
</module>

➜  src cat TestClass.java 
import static java.awt.Button.A;
import java.net.URL; // Violation, wrong order. Caused by static import above; all static imports should be at the bottom of all imports.
import java.security.KeyManagementException;

import javax.net.ssl.TrustManager; // Violation, wrong order and extra separation. 'javax' imports should be above 'java' imports and below 'type' import group with no separation.
import com.google.common.primitives.Doubles.BYTES; // Violation,  wrong order. This import should be above all static and 'java*' imports, in 'type' import group.
public class TestClass { }

➜  src java -jar checkstyle-8.39-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/Tester/src/TestClass.java:2:1: Wrong order for 'java.net.URL' import. [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/TestClass.java:5:1: Extra separation in import group before 'javax.net.ssl.TrustManager' [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/TestClass.java:5:1: Wrong order for 'javax.net.ssl.TrustManager' import. [ImportOrder]
[ERROR] /home/nick/Desktop/Tester/src/TestClass.java:6:1: Wrong order for 'com.google.common.primitives.Doubles.BYTES' import. [ImportOrder]
Audit done.
Checkstyle ends with 4 errors.

```
We cannot change the order of the imports in the config, since this config is made to emulate the import order of intellij.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 3 above)
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:329]: See above comment.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 1 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: i have suggestion you have described a lot in violation comments this will give us error when we do `mvn clean verify` so i was thinking if we add/describe these as bulleted list above the configuration?

Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: I would say just line wrap and ignore the "separation" errors.

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: how about this

```
import static java.awt.Button.A;
import java.net.URL; // Violation, wrong order. Caused by static import above;
import java.security.KeyManagementException;

import javax.swing.JComponent; // Violation, Extra separation and javax should be above java import
import com.neurologic.http.HttpClient; // Violation, should be above static and 'java*' imports

public class TestClass { }
```

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:312]: one last time! pls check this 

```
import static java.awt.Button.A;
import java.net.URL; // Violation, Wrong order, caused by static import above;

import java.security.KeyManagementException; // Violation, Extra separation
import javax.swing.JComponent; // Violation, Wrong order, 'javax' should be above 'java' imports
import com.neurologic.http.HttpClient; // Violation, Should be above static and 'java*' imports

public class TestClass { }
```

**Comment by nmancus1** [comment]
Github, generate site

**Comment by nmancus1** [comment]
Link to check documentation: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/ImportOrderBranch_2021131417/config_imports.html#ImportOrder

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 1 above)
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: Resolved.

**Review by nmancus1 — APPROVED** [review]

**Review by nmancus1 — APPROVED** [review]
Reviewers, please note that this Pr is continuation of https://github.com/checkstyle/checkstyle/pull/9144 and https://github.com/checkstyle/checkstyle/pull/9165.

**Comment by aryaniiit002** [comment]
@strkkk Please review whenever you are free and let me know if there is any change you would like.
I am waiting

**Comment by aryaniiit002** [comment]
Здравствуйте @strkkk
Please review whenever you are free and let me know if there is any change you would like.
I am waiting

**Comment by aryaniiit002** [comment]
Здравствуйте @strkkk
Please review whenever you are free and let me know if there is any change you would like.
I'll wait for you

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 4]
```diff
@@ -140,6 +140,34 @@
...
+ * </p>
+ * <pre>
+ * &lt;module name="ImportOrder"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * import java.io.IOException; // Ok
+ * import java.net.URL; // Ok
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:153]: please change all `Ok` -> `OK` and all `Violation` -> `violation`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 5]
```diff
@@ -140,6 +140,34 @@
...
+ *
+ * import java.io.IOException; // Violation, Extra separation and wrong order
+ *
+ * import javax.net.ssl.SSLContext; // Violation, Extra separation in import group
+ * import javax.net.ssl.TrustManager; // Ok
+ *
+ * import org.apache.http.conn.ClientConnectionManager; // Violation, Extra separation
+ * import org.apache.http.conn.scheme.Scheme; // Ok
+ *
+ * import java.util.Set; // Violation, Extra separation and wrong order
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:163]: there is already example of same vioaltion

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 6]
```diff
@@ -140,6 +140,34 @@
...
+ * &lt;module name="ImportOrder"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * import java.io.IOException; // Ok
+ * import java.net.URL; // Ok
+ *
+ * import java.io.IOException; // Violation, Extra separation and wrong order
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]: `Extra separation and wrong order`
1. Please add clarification where this should be  (wrong order)
2. It is not clear where is extra separation - from top or bottom?
3. Please make violation description lowercase here and below (Extra -> extra)

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 7]
```diff
@@ -297,14 +396,14 @@
  * </pre>
  * <pre>
  * import static java.lang.Math.PI;
- * import static java.lang.Math.abs; // OK, alphabetical case sensitive ASCII order, 'P' &lt; 'a'
- * import static org.abego.treelayout.Configuration.AlignmentInLevel; // OK, alphabetical order
+ * import static java.lang.Math.abs; // Ok, Alphabetical case sensitive ASCII order, 'P' &lt; 'a'
+ * import static org.abego.treelayout.Configuration.AlignmentInLevel; // Ok, Alphabetical order
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:400]: please revert both lines above, it was ok

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 8]
```diff
@@ -375,11 +474,34 @@
  * import static io.netty.handler.codec.http.HttpConstants.COLON;
  * import static io.netty.handler.codec.http.HttpHeaders.addHeader;
  * import static io.netty.handler.codec.http.HttpHeaders.setHeader;
- * import static io.netty.handler.codec.http.HttpHeaders.Names.DATE; // violation
+ * import static io.netty.handler.codec.http.HttpHeaders.Names.DATE; // Violation
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:477]: same here

**Review by strkkk — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 6 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]: i wanted to add but i was getting some error like line length exceed 100
so how should i describe both violation in line 155?

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]: > It is not clear where is extra separation - from top or bottom?

it's always from the top
i guess!

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 5 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:163]: it makes things more clear, like for the 1st code example
should i  remove it?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 6 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]: > i wanted to add but i was getting some error like line length exceed 100
so how should i describe both violation in line 155?

you can wrap comment, like https://checkstyle.org/config_naming.html#CatchParameterName_Examples

>we have config and above config some notes

It is not needed to write a lot, just like `should be in java.io group` or something like this is enough

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]:
```
} catch (Exception FourthException) { // violation, the initial letter
                                          // should be lowercase
      // ...
    }
```
`// ...` this will count as line break

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]: > you can wrap comment

this is giving different CLI output (extra separation violation)

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]: i have a suggestion- instead of writing extra separation or wrong order we can write the property name like
//violation, of property separated and ordered.

Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]: > //violation, of property separated and ordered.

It is better not be cryptic and write it as clear as possible. Message like `// violation, should be in group X` is clearer than mentioning properties

> this is giving different CLI output (extra separation violation)

I think it is fine to have an extra violation if description will be more useful for users

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 5 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:163]: How it is different from line 155? If there is a reason, it is better to explain it in comment for violation
> could you please comment your place time in your next comment

not sure why you need this, but it is 4:56 PM

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:163]: >  it is different from line 155? 
there is no difference except the group changed 
and for the 1st example it shows many different groups separated which should not be separated.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 6 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]: i have a doubt 
ImportOrder [Description](https://checkstyle.org/config_imports.html#ImportOrder_Description)
check's 1st example should be according to description or default values of property from property table?

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]: also some changes in this example

```
import java.io.IOException; // OK
import java.net.URL; // OK

import java.io.IOException; // violation, extra separation before 'java.io.IOException'.
                            // and wrong order for 'java.io.IOException' import.
import javax.swing.JComponent; // violation, extra separation before 'javax.net.ssl.SSLContext'
import javax.net.ssl.TrustManager;
import org.apache.http.conn.ClientConnectionManager; // OK
import java.util.Set; // violation, wrong order for 'java.util.Set' import.
import com.neurologic.http.HttpClient; // violation, wrong order for import
import com.neurologic.http.impl.ApacheHttpClient; // OK

public class SomeClass { }
```

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 5 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: I still think it is not necessary

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 9]
```diff
@@ -140,6 +140,30 @@
...
+ * &lt;module name="ImportOrder"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * import java.io.IOException; // OK
+ * import java.net.URL; // OK
+ *
+ * import java.io.IOException; // violation, extra separation before 'java.io.IOException'.
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:155]: `'java.io.IOException'` no need to add this, it is pretty clear what import this comment is referring to

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 10]
```diff
@@ -140,6 +140,30 @@
...
+ *                             // and wrong order for 'java.io.IOException' import.
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:156]: same here

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 11]
```diff
@@ -140,6 +140,30 @@
...
+ * import javax.swing.JComponent; // violation, extra separation before 'javax.net.ssl.SSLContext'
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:157]: same here

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 12]
```diff
@@ -140,6 +140,30 @@
...
+ * import javax.net.ssl.TrustManager;
+ * import org.apache.http.conn.ClientConnectionManager; // OK
+ * import java.util.Set; // violation, wrong order for 'java.util.Set' import.
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:160]: same here

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 13]
```diff
@@ -208,8 +251,31 @@
...
  * </pre>
  * <p>
- * To configure the check so that it matches default IntelliJ IDEA formatter configuration
- * (tested on v2018.2):
+ * Example:
+ * </p>
+ * <pre>
+ * import static java.io.File.createTempFile;
+ * import static java.lang.Math.abs; // OK, alphabetical case sensitive ASCII order, 'i' &lt; 'l'
+ * import java.lang.Math.sqrt; // OK, follow property 'Option' value 'above'
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:259]: follow -> follows

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 14]
```diff
@@ -251,6 +317,21 @@
...
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * import com.neurologic.http.impl.ApacheHttpClient; // OK
+ * import static java.awt.Button.A;
+ * import javax.swing.JComponent; // violation, wrong order caused by above static import
+ *                      // all static imports comes at bottom
+ * import java.net.URL; // violation, extra separation  in import group before 'java.net.URL'
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:327]: same here, no need to copy import class into comment

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 15]
```diff
@@ -251,6 +317,21 @@
  * &lt;/module&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * import com.neurologic.http.impl.ApacheHttpClient; // OK
+ * import static java.awt.Button.A;
+ * import javax.swing.JComponent; // violation, wrong order caused by above static import
+ *                      // all static imports comes at bottom
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:326]: please align it on same indentation as comment above

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 16]
```diff
@@ -300,11 +393,11 @@
  * import static java.lang.Math.abs; // OK, alphabetical case sensitive ASCII order, 'P' &lt; 'a'
  * import static org.abego.treelayout.Configuration.AlignmentInLevel; // OK, alphabetical order
  *
+ * import java.util.Set; // violation, extra separation in import group before 'java.util.Set'
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:396]: same here, no need to copy class name

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 6 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: >check's 1st example should be according to description or default values of property from property table?

It should show the default config. Yes, it means default values for all properties.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 10 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:156]: so only extra separation before import

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 6 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: so this example is ok or should i change anything?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 11 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:157]: here 157 line 
`extra separation due to above comment` is it okay?

Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:157]: yes

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 10 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:156]: yes

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 11 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 12 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 13 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 15 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 14 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 16 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 9 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 6 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: ok so let me make such example and if u like it we will switch it with the current one. okay?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 16 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: > Are you not on checkstyle gitter?

I am there, but I do not visit it frequently

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 17]
```diff
@@ -268,6 +349,18 @@
...
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * import static java.io.File.createTempFile;
+ * import java.lang.Math.sqrt;
+ *
+ * import javax.swing.JComponent; // violation, extra separation in import group
+ * import static javax.windowConstants.*; // OK
+ *                                     // All static imports are processed like non static imports.
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:360]: please make this comment lowercase

**Review by strkkk — CHANGES_REQUESTED** [review]
last minor

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 17 above)
Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java:360]: u mean All -> all ?

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: done

**Comment by strkkk** [comment]
Github, generate site

**Review by strkkk — APPROVED** [review]

**Comment by aryaniiit002** [comment]
> https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/ImportOrderBranch_2021093618/index.html

it's already looking good thanks to @strkkk and  @nmancus1 helped me, taught me lot of things, answered all my naive questions thank you for that and for your time and also @Abhishek-kumar09 helped me on gitter.
Few more changes then we are good to go i think, so @romani please review and let me know if there is any change you would like.

**Comment by aryaniiit002** [comment]
>https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/ImportOrderBranch_2021093618/config_imports.html#ImportOrder

it's already looking good thanks to @strkkk and  @nmancus1 helped me, taught me lot of things, answered all my naive questions thank you for that and for your time and also @Vyom-Yadav and  @Abhishek-kumar09 helped me on gitter.
Few more changes then we are good to go i think, so @romani please review and let me know if there is any change you would like.

**Review by romani — APPROVED** [review]
